standard/interfaces.md

```diff
@@ -596,9 +512,76 @@ It is an error to declare a class type, struct type, or enum type within the sco
 >
 > *end example*
 
+### §most-specific-implementation Most specific override
+
+Every interface and class shall have a most specific override for every virtual member declared in all interfaces implemented by that type among the overrides appearing in the type or its direct and indirect interfaces. The ***most specific override*** is a unique override that is more specific than every other override. If there is no override, the member itself is considered the most specific override.
+
+> *Note*: The most specific override rule ensures that an ambiguity arising from diamond interface inheritance is resolved explicitly by the programmer at the point where the conflict occurs. *end note*
+
+For a type `T` that is a struct or a class that implements interfaces `I2` and `I3`, where `I2` and `I3` both derive directly or indirectly from interface `I` that declares a member `M`, the most specific override of `M` is:
+
+- If `T` declares an implementation of `I.M`, that implementation is the most specific override.
+- Otherwise, if `T` is a class and a direct or indirect base class declares an implementation of `I.M`, the most derived base class of `T` is the most specific override.
+- Otherwise, if `I2` and `I3` are interfaces implemented by `T` and `I3` derives from `I2` either directly or indirectly, `I3.M` is a more specific implementation than `I2.M`.
+- Otherwise, neither `I2.M` nor `I3.M` are more specific and an error occurs.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods4", expectedErrors:["CS8705"]} -->
+> ```csharp
+> interface IA
+> {
+>     void M() { Console.WriteLine("IA.M"); }
+> }
+> 
+> interface IB : IA
+> {
+>     void IA.M() { Console.WriteLine("IB.M"); }
+> }
+> 
+> interface IC: IA
+> {
+>     void IA.M() { Console.WriteLine("IC.M"); }
+> }
+> 
+> abstract class C: IB, IC { } // error: no most specific override for 'IA.M'
+> 
+> abstract class D: IA, IB, IC // OK
+> {
+>     public abstract void M();
+> }
+> ```
+>
+> The most specific override rule ensures that a conflict (i.e., an ambiguity arising from diamond inheritance) is resolved explicitly by the programmer at the point where the conflict arises. *end example*
+
+It is an error if in a class declaration the most specific override of some interface method is an abstract override that was declared in a class or an interface.
+
+> *Example*:
+>
+> <!-- Example: {template:"standalone-lib-without-using", name:"InterfaceMethods5", expectedErrors:["CS0535"]} -->
+> ```csharp
+> interface IF
+> {
+>     void M();
+> }
+> 
+> abstract class F : IF { } // error: 'F' does not implement 'IF.M'
+> ```
+>
+> *end example*
+
+It is an error if a property `P` has a most specific `get` accessor in one type `T1`, and a most specific `set` accessor in another type `T2`.
```

Comment:
How is it possible for a property `P` has a most specific `get` accessor in one type `T1`, and a most specific `set` accessor in another type `T2`? It is an error for a property to be overridden/explicitly implemented without respecifying all accessors.

Reply:
removed in next commit.